inc/namespace.php

```diff
@@ -586,12 +590,8 @@ function map_user_roles( $user, array $attributes ) {
 			] );
 
 			foreach ( $all_site_ids->sites as $site_id ) {
-				switch_to_blog( $site_id );
-				$user->for_site( $site_id );
-				$user->set_role( reset( $roles['network'] ) );
-
-				foreach ( array_slice( $roles['network'], 1 ) as $role ) {
-					$user->add_role( $role );
+				foreach ( $roles['network'] as $role ) {
+					add_user_to_blog( (int) $site_id, $user->ID, $role );
```

Comment:
In terms of #90, the minimal change here I think would have been to `restore_current_blog()`, although `add_user_to_blog()` might make sense overall. Need to check the difference here.

Also, the previous code skips the first role in `$roles['network']`, why the difference here?